src/hardware/adlib.cpp

```diff
@@ -569,31 +586,73 @@ void Module::DualWrite(uint8_t index, uint8_t port, uint8_t val)
 	CacheWrite(full_port, val);
 }
 
-void Module::CtrlWrite( uint8_t val ) {
-	switch ( ctrl.index ) {
-	case 0x09: /* Left FM Volume */
-		ctrl.lvol = val;
-		goto setvol;
+void Module::CtrlWrite(uint8_t val)
+{
+	switch (ctrl.index) {
+	case 0x04:
+		if (adlib_gold) {
+			adlib_gold->StereoControlWrite(StereoProcessorControlReg::VolumeLeft,
+			                               val);
+		}
+		break;
+	case 0x05:
+		if (adlib_gold) {
+			adlib_gold->StereoControlWrite(StereoProcessorControlReg::VolumeRight,
+			                               val);
+		}
+		break;
+	case 0x06:
+		if (adlib_gold) {
+			adlib_gold->StereoControlWrite(StereoProcessorControlReg::Bass,
+			                               val);
+		}
+		break;
+	case 0x07:
+		if (adlib_gold) {
+			adlib_gold->StereoControlWrite(StereoProcessorControlReg::Treble,
+			                               val);
+		}
+		break;
+
+	case 0x08:
+		if (adlib_gold) {
+			adlib_gold->StereoControlWrite(
+			        StereoProcessorControlReg::SwitchFunctions, val);
+		}
+		break;
+
+	case 0x09: /* Left FM Volume */ ctrl.lvol = val; goto setvol;
 	case 0x0a: /* Right FM Volume */
 		ctrl.rvol = val;
-setvol:
-		if ( ctrl.mixer ) {
-			//Dune cdrom uses 32 volume steps in an apparent mistake, should be 128
-			mixerChan->SetVolume( (float)(ctrl.lvol&0x1f)/31.0f, (float)(ctrl.rvol&0x1f)/31.0f );
+	setvol:
+		if (ctrl.mixer) {
```

Comment:
Thinking how we need to set `sbmixer = false` when using AdlibGold, if this is always the case, should be block the SB mixer programmatically when in adlib-gold mode?

(I think there's a searate CT mixer handlers in sblaster as well - perhaps we should disable those?)

Reply:
I think it's a good idea; AdLib Gold emulation is useful for Dune only, after all.

Reply:
After further consideration I won't disable this. That would effectively disable the in-game mixer for the music in the CD version, which doesn't feel quite right, and there's also a handy feature to automatically lower the FM volume during dialogs and even subtly change the panning (see defaults on the screenshot below). That's a feature we definitely don't want to hard-disable.

Furthermore, I'd argue the CD version is the definitive version of the game anyway, and that doesn't clip. You only need to mess around with the mixer manually with the floppy version.

![image](https://user-images.githubusercontent.com/698770/174477602-44481ce9-4d09-413c-89b1-38629af76534.png)